toolkit/Makefile

```diff
@@ -141,7 +141,24 @@ include $(SCRIPTS_DIR)/utils.mk
 
 ######## REMAINING BUILD FLAGS ########
 
+##help:var:DAILY_BUILD_ID:{lkg,lkg-force,<build_id>}='lkg' will auto select latest daily build repo, 'lkg-force' will also replace the toolchain manifest files. An explicit ID of the daily build repo can also be used.
 ifneq ($(DAILY_BUILD_ID),)
+   # If DAILY_BUILD_ID is set to "lkg" or "lkg-force" we should attempt to auto select the best daily build
+   ifeq ($(DAILY_BUILD_ID),lkg)
+      $(warning Auto detecting DAILY_BUILD_ID based on the latest known good build)
+      override DAILY_BUILD_ID := $(shell $(SCRIPTS_DIR)/setuplkgtoolchain.sh --auto-id)
+      ifneq ($(.SHELLSTATUS),0)
+         $(error Failed to auto detect DAILY_BUILD_ID... try DAILY_BUILD_ID=lkg-force to overwrite manifests)
+      endif
+   endif
+   ifeq ($(DAILY_BUILD_ID),lkg-force)
+      $(info Auto detecting DAILY_BUILD_ID based on the latest known good build AND overwriting manifests)
+      override DAILY_BUILD_ID := $(shell $(SCRIPTS_DIR)/setuplkgtoolchain.sh --auto-id --force)
+      ifneq ($(.SHELLSTATUS),0)
+         $(error Failed to auto detect DAILY_BUILD_ID and update manifests)
+      endif
+   endif
+
```

Comment:
I have nothing against adding an interface to run `setuplkgtoolchain.sh` directly from a `make` command. I don't want to overload `DAILY_BUILD_ID`, however. With the current change we're adding special values modifying what setting `DAILY_BUILD_ID` does, which is modifying the URL lists to look for the toolchain packages under a different address. It's a different responsibility, so I think we should add a separate argument for this behaviour.

Reply:
I don't really see a problem with this change. I think it's fine to overload the DAILY_BUILD_ID value with `lkg` or `lkg-force`. Just my opinion, but I think we have too many make targets already, and we should try to consolidate them and avoid creating new ones unless they are really necessary.

Reply:
Was going to respond the same, I'd like to try and consolidate our flags more in the future. Its much easier to offer help/readme type documentation for a smaller number of commands that each do a few things. In this case I feel like the explicit and automatic modes are doing something VERY similar.